src/librustc_typeck/collect.rs

```diff
@@ -2583,6 +2586,35 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
     codegen_fn_attrs
 }
 
+/// Checks if the provided DefId is a method in a trait impl for a trait which has track_caller 
+/// applied to the method prototype.
+fn should_inherit_track_caller(tcx: TyCtxt<'_>, def_id: DefId) -> bool {
+    if let Some(assoc) = tcx.opt_associated_item(def_id) {
+        if let ty::AssocItemContainer::ImplContainer(container) = assoc.container {
+            if let Some(trait_id) = tcx.trait_id_of_impl(container) {
+                if let Some(trait_method_id) = tcx
+                    .associated_items(trait_id)
+                    .filter_by_name_unhygienic(assoc.ident.name)
+                    .find(move |impl_item| match impl_item.kind {
+                        ty::AssocKind::Method => {
+                            tcx.hygienic_eq(impl_item.ident, assoc.ident, trait_id)
+                        }
+                        _ => false,
+                    })
+                    .copied()
```

Comment:
Shouldn't need `.copied()`, AFAICT the element type is a `&'tcx ty::AssocItem`.

Reply:
it's my fault bc i `.copied()` from the example query in the specialization graph you showed me!